.tekton/tasks/promote-component.yaml

```diff
@@ -0,0 +1,276 @@
+apiVersion: tekton.dev/v1
+kind: Task
+metadata:
+  labels:
+    app.kubernetes.io/version: "0.1"
+  annotations:
+    tekton.dev/pipelines.minVersion: "0.12.1"
+    tekton.dev/tags: "appstudio, hacbs"
+  name: promote-component
+spec:
+  description: |
+    Clones redhat-appstudio/infra-deployments repository, runs script in 'SCRIPT' parameter, generates pull-request for redhat-appstudio/infra-deployments repository.
+  params:
+    - name: SCRIPT
+      description: Bash script for changing the infra-deployments
+    - name: BRANCH_NAME
+      description: Source branch in infra-deployment repo
+    - name: COMPONENT_NAME
+      description: Component to promote
+    - name: GH_REPO
+      description: GitHub repository of the infra-deployments code
+      default: redhat-appstudio/infra-deployments
+    - name: GIT_IMAGE
+      description: Image reference containing the git command
+      default: registry.redhat.io/openshift-pipelines/pipelines-git-init-rhel8:v1.8.2-8@sha256:a538c423e7a11aae6ae582a411fdb090936458075f99af4ce5add038bb6983e8
+    # per https://kubernetes.io/docs/concepts/containers/images/#imagepullpolicy-defaulting
+    # the cluster will set imagePullPolicy to IfNotPresent
+     # also per direction from Ralph Bean, we want to use image digest based tags to use a cue to automation like dependabot or renovatebot to periodially submit pull requests that update the digest as new images are released.
+    - name: SCRIPT_IMAGE
+      description: Image reference for SCRIPT execution
+      # this image is built using https://github.com/redhat-appstudio/build-tasks-dockerfiles/blob/main/update-infra-deployments-task-scripts-image/Dockerfile
+      default: quay.io/redhat-appstudio/update-infra-deployments-task-script-image@sha256:2748f1a4f1af4e35214745aed4e56a9d06f6bdbd30572e7ade13729e67f23cc9
```

Comment:
Maybe we can use the `update-infra-deployments-task-script-image` as the GIT_IMAGE as well? It looks like it contains `git`. That way we wouldn't have to depend on the pipelines-git-init image

Reply:
Good point. Merged two parameters `GIT_IMAGE` and `SCRIPT_IMAGE` into one parameter `IMAGE` now and using `update-infra-deployments-task-script-image` for both the steps.